zerver/middleware.py

```diff
@@ -289,7 +289,10 @@ def process_response(self, request: HttpRequest,
         try:
             requestor_for_logs = request._requestor_for_logs
         except Exception:
-            requestor_for_logs = "unauth"
+            if hasattr(request, 'user') and hasattr(request.user, 'format_requestor_for_logs'):
+                requestor_for_logs = request.user.format_requestor_for_logs()
+            else:
+                requestor_for_logs = "unauth"
```

Comment:
I think we might want to make `requestor_for_logs` in the unauth case still show the subdomain, i.e. `string_id/unauth`?